imgparse/imgparse.py

```diff
@@ -473,16 +516,20 @@ def get_gsd(image_path, exif_data=None, xmp_data=None, corrected_alt=None):
     :param exif_data: the exif dictionary for the image (optional to speed up processing)
     :param xmp_data: the XMP data of image, as a string dump of the original XML (optional to speed up processing)
     :param corrected_alt: corrected relative altitude (optional)
+    :param use_calibrated_focal_length: enable to use calibrated focal length if available
+    :param use_rlf: enable to, if session_alt is not enabled, use rlf data, if available, for Sentera imagery
     :return: **gsd** - the ground sample distance of the image in meters
     :raises: ParsingError
     """
-    focal, pitch = get_camera_params(image_path, exif_data)
+    if not xmp_data:
+        xmp_data = get_xmp_data(image_path)
```

Comment:
Cleaner if you use the `get_if_needed()` decorator

Reply:
Addressed